### ISSUE TITLE
spring actuator is causing consumer to log "Found no committed offset for partition"

### ISSUE THREAD

**Issue by AttitudeL** [spring-cloud/spring-cloud-stream#2223]
Dear developers,

I recently discovered some weird behaviour from the log of my application. One of the consumer group is logging many `Found no committed offset for partition` from the consumers inside that group. So I created a reproducible project and below you can see the code in details.

**application.yml**
```
server:
  port: 8080
  address: 127.0.0.1

spring.cloud.stream:
  function:
    definition: retryProcessor
  kafka.binder:
    brokers: localhost:9092
    autoCreateTopics: false
    configuration:
      # Consumer
      auto.offset.reset: earliest
      # Producer
      acks: all
      linger.ms: 0
      batch.size: 0
      max.in.flight.requests.per.connection: 1
  kafka.bindings:
    retryProcessor-in-0.consumer:
      autoCommitOffset: false
      idleEventInterval: 1000
  bindings:
    retryProcessor-in-0:
      group: retry-group
      destination: retry_topic
      consumer:
        maxAttempts: 1
        concurrency: 12

spring.cloud.stream.kafka.streams:
  binder:
    brokers: localhost:9092
    deserializationExceptionHandler: logAndContinue
    autoCreateTopics: false
    configuration:
      # General
      commit.interval.ms: 1000
      replication.factor: 3
      num.stream.threads: 12
      default.key.serde: org.apache.kafka.common.serialization.Serdes$StringSerde
      default.value.serde: org.apache.kafka.common.serialization.Serdes$StringSerde
      state.dir: /tmp
      # Consumer
      auto.offset.reset: earliest
      # Producer
      acks: all
      linger.ms: 0
      batch.size: 0
      max.in.flight.requests.per.connection: 1
```

**DemoApplication.class**
```
package com.example.demo;

import org.apache.kafka.clients.consumer.Consumer;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
import org.springframework.kafka.support.Acknowledgment;
import org.springframework.kafka.support.KafkaHeaders;
import org.springframework.messaging.Message;
import org.springframework.messaging.MessageHeaders;

@SpringBootApplication
public class DemoApplication {

	private static final Logger log = LogManager.getLogger(DemoApplication.class);

	public static void main(String[] args) {
		SpringApplication.run(DemoApplication.class, args);
	}

	@Bean
	public java.util.function.Consumer<Message<String>> retryProcessor() {
		return message -> {
			MessageHeaders headers = message.getHeaders();
			String payload = message.getPayload();
			Consumer<?, ?> consumer = headers.get(KafkaHeaders.CONSUMER, Consumer.class);
			String topic = headers.get(KafkaHeaders.RECEIVED_TOPIC, String.class);
			Integer partitionId = headers.get(KafkaHeaders.RECEIVED_PARTITION_ID, Integer.class);
			Acknowledgment acknowledgment = headers.get(KafkaHeaders.ACKNOWLEDGMENT, Acknowledgment.class);

			log.info("Topic {}, Partition ID {}, Payload {}", topic, partitionId, payload);

			// pause consumer for certain period of time before retry

			// retry payload delivery
			acknowledgment.acknowledge();
		};
	}
}
```

**pom.xml**
```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <parent>
    <groupId>org.springframework.boot</groupId>
    <artifactId>spring-boot-starter-parent</artifactId>
    <version>2.5.4</version>
    <relativePath/> <!-- lookup parent from repository -->
  </parent>

  <groupId>com.example</groupId>
  <artifactId>demo</artifactId>
  <version>0.0.1-SNAPSHOT</version>
  <name>demo</name>
  <description>demo</description>

  <properties>
    <java.version>1.8</java.version>
    <spring-cloud.version>2020.0.3</spring-cloud.version>
  </properties>

  <dependencies>
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter</artifactId>
      <exclusions>
        <exclusion>
          <groupId>org.springframework.boot</groupId>
          <artifactId>spring-boot-starter-logging</artifactId>
        </exclusion>
      </exclusions>
    </dependency>
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-web</artifactId>
    </dependency>
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-validation</artifactId>
    </dependency>
    <dependency>
      <groupId>org.springframework.cloud</groupId>
      <artifactId>spring-cloud-stream-binder-kafka</artifactId>
    </dependency>
    <dependency>
      <groupId>org.springframework.cloud</groupId>
      <artifactId>spring-cloud-stream-binder-kafka-streams</artifactId>
    </dependency>
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-log4j2</artifactId>
    </dependency>
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-actuator</artifactId>
    </dependency>

    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-test</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>org.springframework.cloud</groupId>
      <artifactId>spring-cloud-stream</artifactId>
      <scope>test</scope>
      <classifier>test-binder</classifier>
      <type>test-jar</type>
    </dependency>
  </dependencies>
  <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>org.springframework.cloud</groupId>
        <artifactId>spring-cloud-dependencies</artifactId>
        <version>${spring-cloud.version}</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
    </dependencies>
  </dependencyManagement>

  <build>
    <plugins>
      <plugin>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-maven-plugin</artifactId>
      </plugin>
    </plugins>
  </build>

</project>
```

When the application is started it periodically logging `Found no committed offset for partition`.
The following is a portion of the log showing such messages:
```
2021-09-17 10:45:42.109  INFO 19693 --- [container-3-C-1] o.a.k.c.c.i.ConsumerCoordinator          : [Consumer clientId=consumer-retry-group-5, groupId=retry-group] Found no committed offset for partition retry_topic-15
2021-09-17 10:45:42.109  INFO 19693 --- [container-1-C-1] o.a.k.c.c.i.ConsumerCoordinator          : [Consumer clientId=consumer-retry-group-3, groupId=retry-group] Found no committed offset for partition retry_topic-10
2021-09-17 10:45:42.109  INFO 19693 --- [container-3-C-1] o.a.k.c.c.i.ConsumerCoordinator          : [Consumer clientId=consumer-retry-group-5, groupId=retry-group] Found no committed offset for partition retry_topic-14
2021-09-17 10:45:42.110  INFO 19693 --- [container-6-C-1] o.a.k.c.c.i.ConsumerCoordinator          : [Consumer clientId=consumer-retry-group-8, groupId=retry-group] Found no committed offset for partition retry_topic-21
2021-09-17 10:45:42.110  INFO 19693 --- [container-6-C-1] o.a.k.c.c.i.ConsumerCoordinator          : [Consumer clientId=consumer-retry-group-8, groupId=retry-group] Found no committed offset for partition retry_topic-20
2021-09-17 10:45:42.110  INFO 19693 --- [container-5-C-1] o.a.k.c.c.i.ConsumerCoordinator          : [Consumer clientId=consumer-retry-group-7, groupId=retry-group] Found no committed offset for partition retry_topic-19
2021-09-17 10:45:42.110  INFO 19693 --- [ontainer-10-C-1] o.a.k.c.c.i.ConsumerCoordinator          : [Consumer clientId=consumer-retry-group-12, groupId=retry-group] Found no committed offset for partition retry_topic-5
2021-09-17 10:45:42.110  INFO 19693 --- [ontainer-10-C-1] o.a.k.c.c.i.ConsumerCoordinator          : [Consumer clientId=consumer-retry-group-12, groupId=retry-group] Found no committed offset for partition retry_topic-4
2021-09-17 10:45:42.110  INFO 19693 --- [container-0-C-1] o.a.k.c.c.i.ConsumerCoordinator          : [Consumer clientId=consumer-retry-group-2, groupId=retry-group] Found no committed offset for partition retry_topic-9
2021-09-17 10:45:42.111  INFO 19693 --- [container-9-C-1] o.a.k.c.c.i.ConsumerCoordinator          : [Consumer clientId=consumer-retry-group-11, groupId=retry-group] Found no committed offset for partition retry_topic-3
2021-09-17 10:45:42.111  INFO 19693 --- [container-9-C-1] o.a.k.c.c.i.ConsumerCoordinator          : [Consumer clientId=consumer-retry-group-11, groupId=retry-group] Found no committed offset for partition retry_topic-2
2021-09-17 10:45:42.111  INFO 19693 --- [container-2-C-1] o.a.k.c.c.i.ConsumerCoordinator          : [Consumer clientId=consumer-retry-group-4, groupId=retry-group] Found no committed offset for partition retry_topic-13
2021-09-17 10:45:42.111  INFO 19693 --- [container-2-C-1] o.a.k.c.c.i.ConsumerCoordinator          : [Consumer clientId=consumer-retry-group-4, groupId=retry-group] Found no committed offset for partition retry_topic-12
```

I was playing around with different config for kafka, no luck, until I removed `spring-boot-starter-actuator` from pom.
After removing `spring-boot-starter-actuator`, I no longer see such logs showing and I've waited for an hour and still no such logs. So I don't know why adding actuator would cause the consumer to keep on logging `Found no committed offset for partition`. Is this a known issue?

**Comment by bono007** [comment]
Probably [this](https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/pull/1123). Removing Actuator would also remove metrics - so it lines up.

**Comment by AttitudeL** [comment]
> Probably [this](https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/pull/1123). Removing Actuator would also remove metrics - so it lines up.

Thanks for the info, it does look like the same problem. However the issue you linked has been marked as resolved. I'm on the latest dependencies for spring boot and spring cloud stream and still seeing such logs. I wonder if the changes were actually fixing the log noises.

**Comment by bono007** [comment]
> > Probably [this](https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/pull/1123). Removing Actuator would also remove metrics - so it lines up.
> 
> Thanks for the info, it does look like the same problem. However the issue you linked has been marked as resolved. I'm on the latest dependencies for spring boot and spring cloud stream and still seeing such logs. I wonder if the changes were actually fixing the log noises.

Hi @AttitudeL,

>  latest dependencies for spring boot and spring cloud stream

So the dependency that would have the fix is `spring-cloud-stream-binder-kafka`. However, the fix was committed Aug 18 but the project's latest milestone version that would contain the fix (`3.2.0-M1`) was on [Jul 30](https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/releases/tag/v3.2.0-M1). You will need to use the next published version. I usually can tell about when that will be based on https://calendar.spring.io/ . But I do not see the next Spring Cloud Stream 3.2.x due date. 

Until then I would adjust my log level as [suggested here](https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/pull/1123#issuecomment-902909179). I have done that as well via [application.yml](https://docs.spring.io/spring-boot/docs/2.4.8/reference/htmlsingle/#boot-features-custom-log-levels).

**Comment by AttitudeL** [comment]
Verified this issue is a duplicate of https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/issues/1107. Currently suggested fix is to reduce the log level for `o.a.k.c.c.internals.ConsumerCoordinator` to WARN.
